tests/structure/io/test_pdbx.py

```diff
@@ -214,6 +214,31 @@ def test_metal_coordination_bonds():
     assert np.all(conn_type_id == "metalc")
 
 
+def test_connect_via_residue_names_with_alt_atom_ids():
+    """
+    Ensure that the residue names are correctly parsed, even when alternative
+    atom IDs are present.
+    """
+    pdbx_file_with_no_alt_ids = pdbx.CIFFile.read(
+        join(data_dir("structure"), "6q9t.cif")
+    )
+    atoms = pdbx.get_structure(pdbx_file_with_no_alt_ids, model=1, include_bonds=True)
+
+    # ZY9 modified to have alternative atom IDs
+    pdbx_file_with_alt_ids = pdbx.CIFFile.read(
+        join(data_dir("structure"), "6q9t_with_alt_ids.cif")
+    )
+    atoms_with_alt_ids = pdbx.get_structure(
+        pdbx_file_with_alt_ids, model=1, include_bonds=True
+    )
```

Comment:
`connect_via_residue_names()` is not specific to PDBx files, but can be used on any `AtomArray`. Hence, I propose to move such test to `test_bonds.py`. This would also remove the need to introduce a new test structure as you could run this test on `ZY9` via `structure.info.residue("ZY9")` directly, where its atom names are modified before passing it to `connect_via_residue_names()`